install/installer/scripts/kots-install.sh

```diff
@@ -0,0 +1,345 @@
+#!/bin/sh
+# Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+# Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+
+# shellcheck disable=SC2050,SC2153
+
+set -e
+
+echo "Gitpod: Killing any in-progress installations"
+
+kubectl delete jobs.batch -n "${NAMESPACE}" -l component="gitpod-installer,cursor!=${CURSOR}" --force --grace-period 0 || true
+kubectl delete pod -n "${NAMESPACE}" -l component="gitpod-installer,cursor!=${CURSOR}" --force --grace-period 0 || true
+
+if [ "$(helm status -n "${NAMESPACE}" gitpod -o json | jq '.info.status == "deployed"')" = "false" ];
+then
+    echo "Gitpod: Deployment in-progress - clearing"
+
+    VERSION="$(helm status -n "${NAMESPACE}" gitpod -o json | jq '.version')"
+    if [ "${VERSION}" -le 1 ];
+    then
+        echo "Gitpod: Uninstall application"
+        helm uninstall -n "${NAMESPACE}" gitpod --wait || true
+    else
+        echo "Gitpod: Rolling back application"
+        helm rollback -n "${NAMESPACE}" gitpod --wait || true
+    fi
+fi
+
+echo "Gitpod: Create a Helm template directory"
+rm -Rf "${GITPOD_OBJECTS}"
+mkdir -p "${GITPOD_OBJECTS}/templates"
+cat <<EOF >> "${GITPOD_OBJECTS}/Chart.yaml"
+apiVersion: v2
+name: gitpod-kots
+description: Always ready-to-code
+version: "1.0.0"
+appVersion: "$(/app/installer version | yq e '.version' -)"
+EOF
+
+echo "Gitpod: Generate the base Installer config"
+/app/installer init > "${CONFIG_FILE}"
+
+echo "Gitpod: auto-detecting ShiftFS support on host machine"
+kubectl wait job -n "${NAMESPACE}" --for=condition=complete -l component=shiftfs-module-loader --timeout=30s || true
+ENABLE_SHIFTFS=$(kubectl get jobs.batch -n "${NAMESPACE}" -l component=shiftfs-module-loader -o jsonpath='{.items[0].status.succeeded}')
+
+if [ "${ENABLE_SHIFTFS}" = "1" ]; then
+    echo "Gitpod: enabling ShiftFS support"
+
+    yq e -i '.workspace.runtime.fsShiftMethod = "shiftfs"' "${CONFIG_FILE}"
+fi
+
+echo "Gitpod: auto-detecting containerd location on host machine"
+if [ -d "/mnt/node0${CONTAINERD_DIR_K3S}" ]; then
+    echo "Gitpod: containerd dir detected as k3s"
+
+    yq e -i ".workspace.runtime.containerdRuntimeDir = \"${CONTAINERD_DIR_K3S}\"" "${CONFIG_FILE}"
+elif [ -d "/mnt/node0${CONTAINERD_DIR_AL}" ]; then
+    echo "Gitpod: containerd dir detected as ${CONTAINERD_DIR_AL}"
+
+    yq e -i ".workspace.runtime.containerdRuntimeDir = \"${CONTAINERD_DIR_AL}\"" "${CONFIG_FILE}"
+fi
+
+if [ -S "/mnt/node0${CONTAINERD_SOCKET_K3S}" ]; then
+    echo "Gitpod: containerd socket detected as k3s"
+
+    yq e -i ".workspace.runtime.containerdSocket = \"${CONTAINERD_SOCKET_K3S}\"" "${CONFIG_FILE}"
+elif [ -S "/mnt/node0${CONTAINERD_SOCKET_AL}" ]; then
+    echo "Gitpod: containerd socket detected as ${CONTAINERD_SOCKET_AL}"
+
+    yq e -i ".workspace.runtime.containerdSocket = \"${CONTAINERD_SOCKET_AL}\"" "${CONFIG_FILE}"
+fi
+
+echo "Gitpod: Inject the Replicated variables into the config"
```

Comment:
*Definitely not asking in this PR*, but a next iteration of this change would be to move all the config settings into a separate script so that we can validate the end config as a pre-flight check right? Should be simple enough and hence solving https://github.com/gitpod-io/gitpod/issues/12015 :thinking:

Reply:
That's exactly the next step - good thought